bin/cli.js

```diff
@@ -0,0 +1,233 @@
+  config.logLevel = karma.constants[karmaLogLevels[logLevel][0]];
+  config.webpackMiddleware.stats = karmaLogLevels[logLevel][1];
+};
+
+const removeNodePackages = (config, ...packages) => {
+  if (!config.webpack) config.webpack = {};
+  if (!config.webpack.node) config.webpack.node = {};
+  packages.forEach(lib => {
+    config.webpack.node[lib] = 'empty';
+  });
+};
+
+const getReporter = logLevel => {
+  const browsers = [];
+  const reporter = function(base) {
+    base(this);
+    if (logLevel === 'quiet') {
+      this.onBrowserError = () => {};
+      this.onBrowserLog = () => {};
+      return;
+    }
+
+    this.onBrowserLog = (browser, log) => {
+      if (typeof log !== 'string') log = JSON.stringify(log);
+      else log = log.slice(1, log.length - 1);
+      if (!browsers.includes(browser)) {
+        browsers.push(browser);
+        console.log(`\n${browser}:`);
+      }
+      console.log(log);
+    };
+  };
+
+  reporter.$inject = ['baseReporterDecorator'];
+  return reporter;
+};
+
+const getBrowserConfig = conf => {
+  const config = {
+    preprocessors: {},
+    files: [],
+    plugins: [
+      'karma-webpack',
+      { 'reporter:meta': ['type', getReporter(conf.browser.logLevel)] },
+    ],
+    reporters: [ 'meta' ],
+    basePath: process.env.PWD,
+    port: conf.browser.port,
+    autoWatch: false,
+    singleRun: true,
+    concurrency: 1,
+  };
+
+  const adapter = path.resolve('./build/adapter.js');
+  config.files.push(adapter);
+  config.preprocessors[adapter] = ['webpack'];
+  const nodePackages = [ 'fs', 'child_process' ];
+
+  removeNodePackages(config, ...nodePackages);
+  setKarmaBrowsers(config, ...conf.browser.browsers);
+  setKarmaLogLevel(config, conf.browser.logLevel);
+  return config;
+};
+
+const getConfig = () => {
+  const version = parseFile(path.resolve(__dirname, '../package.json')).version;
+  program.version(version).usage('[options] -- <file ...>');
+  cliOptions.forEach(option => program.option(...option));
+  program.parse(process.argv);
+
+  const config = program.config ? parseFile(program.config) : {};
+
+  config.files = merge(config.files, program.args);
+  config.files = loadFiles(config.files);
+  config.exclude = merge(config.exclude, program.exclude);
+  config.files = exclude(config.files, config.exclude);
+  config.reporter = program.logLevel || program.reporter || 'default';
+
+  config.browser = config.browser || {};
+  config.browser.browsers = merge(config.browser.browsers, program.browsers);
+  config.browser.logLevel = program.logLevel || config.browser.logLevel;
+
+  config.browser.port = +program.browserPort || config.browser.port;
+  config.browser = getBrowserConfig(config);
+  return config;
+};
+
+const runBrowser = (config, cb) => {
+  if (!config.browser.browsers.length) {
+    console.error('No browser environments specified');
+    cb(1);
+  }
+
+  const buildDir = path.resolve('./build');
+  const buildAdapter = path.resolve('./build/adapter.js');
```

Comment:
This will use current directory as `process.cwd()`, therefore if running from i.e. test/ this will resolve to test/build and test/adapter.js and I don't this that's appropriate. You should use `__dirname` or other methods here and in other places.